zerver/data_import/mattermost.py

```diff
@@ -389,13 +425,21 @@ def message_to_dict(post_dict: Dict[str, Any]) -> Dict[str, Any]:
         else:
             reactions = []
 
-        return dict(
+        message_dict = dict(
             sender_id=sender_id,
-            receiver_id=post_dict["channel"],
             content=content,
             pub_date=int(post_dict['create_at'] / 1000),
             reactions=reactions
         )
+        if "channel" in post_dict:
+            message_dict["channel_name"] = post_dict["channel"]
+        elif "channel_members" in post_dict:
+            channel_members = post_dict["channel_members"]
+            if len(channel_members) > 2:
+                message_dict["huddle_name"] = generate_huddle_name(channel_members)
+            elif len(channel_members) == 2:
+                message_dict["pm_members"] = channel_members
+        return message_dict
```

Comment:
Should there be an `else raise AssertionError` here, or is it possible to have posts without `channel_members` in it?

Reply:
Added AssertionError.direct_posts should have `channel_members` in them.